src/daemon/rpmostreed-transaction-types.c

```diff
@@ -2060,6 +2060,158 @@ rpmostreed_transaction_new_refresh_md (GDBusMethodInvocation *invocation,
 
 }
 
+/* ================================ ModifyYumRepo ================================ */
+
+typedef struct {
+  RpmostreedTransaction parent;
+  char *osname;
+  char *repo_id;
+  GVariant *repo_options;
+} ModifyYumRepoTransaction;
+
+typedef RpmostreedTransactionClass ModifyYumRepoTransactionClass;
+
+GType modify_yum_repo_transaction_get_type (void);
+
+G_DEFINE_TYPE (ModifyYumRepoTransaction,
+               modify_yum_repo_transaction,
+               RPMOSTREED_TYPE_TRANSACTION)
+
+static void
+modify_yum_repo_transaction_finalize (GObject *object)
+{
+  ModifyYumRepoTransaction *self;
+
+  self = (ModifyYumRepoTransaction *) object;
+  g_free (self->osname);
+  g_free (self->repo_id);
+  g_variant_unref (self->repo_options);
+
+  G_OBJECT_CLASS (modify_yum_repo_transaction_parent_class)->finalize (object);
+}
+
+static DnfRepo *
+get_dnf_repo_by_id (RpmOstreeContext *ctx, const char *repo_id)
+{
+  DnfContext *dnfctx = rpmostree_context_get_dnf (ctx);
+
+  GPtrArray *repos = dnf_context_get_repos (dnfctx);
+  for (guint i = 0; i < repos->len; i++)
+    {
+      DnfRepo *repo = repos->pdata[i];
+      if (g_strcmp0 (dnf_repo_get_id (repo), repo_id) == 0)
+        return repo;
+    }
+
+  return NULL;
+}
+
+static gboolean
+modify_yum_repo_transaction_execute (RpmostreedTransaction *transaction,
+                                     GCancellable *cancellable,
+                                     GError **error)
+{
+  ModifyYumRepoTransaction *self = (ModifyYumRepoTransaction *) transaction;
+  OstreeSysroot *sysroot = rpmostreed_transaction_get_sysroot (transaction);
+
+  if (self->repo_id == NULL)
+    return glnx_throw (error, "Invalid repo_id");
+
+  if (self->repo_options == NULL)
+    return glnx_throw (error, "Invalid repo_options");
```

Comment:
I think we can drop these checks (or if you'd like turn them into `g_assert`) since IIRC we're pretty much guaranteed values by D-Bus not supporting `NULL`, right?

Reply:
Done, removed those.